src/utils/Authorizations.ts

```diff
@@ -0,0 +1,98 @@
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function hasAuthzRole (role: string): boolean {
+  return !!store.getAuthorizations.roles.find(r => r === role)
+}
+
+/** Whether the current account has the given Keycloak role. */
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function hasKeycloakRole (role: string): boolean {
+  return !!store.getKeycloakRoles.find(r => r === role)
+}
+
+/** Whether the user is Business Registry Staff (which is not the same as Service BC Staff). */
+function isRegStaff (): boolean {
+  return store.getKeycloakRoles.includes(KeycloakRoles.STAFF)
+}
+
+/** Whether the user is Service BC Staff (which is not the same as Business Registry Staff). */
+function isSbcStaff (): boolean {
+  return (store.getAccountInformation.accountType === AccountTypes.SBC_STAFF)
+}
+
+/** Whether the current account is a Premium account. */
+function isPremiumAccount (): boolean {
+  return (store.getAccountInformation.accountType === AccountTypes.PREMIUM)
+}
+
+/**
+ * Returns True if the specified action is authorized, else False.
+ * @param action the action to check
+ */
+export function IsAuthorized (action: AuthorizedActions): boolean {
+  switch (action) {
+    case AuthorizedActions.FOLIO_NUMBER: return isPremiumAccount()
+    case AuthorizedActions.SBC_BREADCRUMBS: return isSbcStaff()
+    case AuthorizedActions.STAFF_BREADCRUMBS: return isRegStaff()
+    case AuthorizedActions.STAFF_COMMENTS: return isRegStaff()
+    case AuthorizedActions.BLANK_COMPLETING_PARTY: return (isSbcStaff() || isRegStaff())
+    case AuthorizedActions.BLANK_CERTIFY_STATE: return (isSbcStaff() || isRegStaff())
+    case AuthorizedActions.EDITABLE_EMAIL_ADDRESS: return (isSbcStaff() || isRegStaff())
+    case AuthorizedActions.EDITABLE_COMPLETING_PARTY: return (isSbcStaff() || isRegStaff())
+    case AuthorizedActions.EDITABLE_CERTIFY_NAME: return (isSbcStaff() || isRegStaff())
+    case AuthorizedActions.THIRD_PARTY_CERTIFY_STMT: return (isSbcStaff() || isRegStaff())
+    case AuthorizedActions.NO_COMPLETING_PARTY_MESSAGE_BOX: return isRegStaff()
+    case AuthorizedActions.NO_MIN_START_DATE: return isRegStaff()
+    case AuthorizedActions.NO_CONTACT_INFO: return isRegStaff()
+    case AuthorizedActions.COURT_ORDER_POA: return isRegStaff()
+    case AuthorizedActions.FIRM_COMPLETING_PARTY: return isRegStaff()
+    case AuthorizedActions.STAFF_PAYMENT: return isRegStaff()
+    case AuthorizedActions.FILE_AND_PAY: return !isSbcStaff()
+    case AuthorizedActions.AML_OVERRIDES: return isRegStaff()
+    case AuthorizedActions.ADD_EDIT_BUSINESS: return isRegStaff()
+    case AuthorizedActions.ADD_UNREGISTERED_BUSINESS: return isRegStaff()
+    case AuthorizedActions.ADD_REGISTERED_BUSINESS: return isRegStaff()
+    case AuthorizedActions.CONTINUATION_IN_FILING: return isRegStaff()
+    case AuthorizedActions.RESTORATION_FILING: return isRegStaff()
+  }
+  return false // should never happen
+}
```

Comment:
OK, for the record, this code is temporary. It is an intermediate step to refactor all the code that previously said "if staff" to now say "if authorized".

The end goal is to have the permissions (aka authorized actions) passed down to the UI from the Keycloak token (or Authorizations call -- same list). In other words, the role-permissions mapping would be done in KC config, not in the UI. This would allow the creation of new user types, or the tailoring of roles, in external config instead of UI code.

Thoughts?

cc: @ochiu @seeker25

Reply:
looks and sounds good to me

Reply:
Authorizations call, I think it's the same as looking at the realm roles in the JWT as well?

Reply:
I think eventually we were going to put everything in the JWT (but that's in the distant future) where the OIDC would call into AUTH and then include all those details in the JWT. We can't do that right now on shared services keycloak because we can't build any custom plugins.

Reply:
The roles list from the `authorizations` call is a superset of the JWT group roles. It also includes row-level permissions.